tests/extension/oneapi_root_group/root_group.cpp

```diff
@@ -0,0 +1,256 @@
+  REQUIRE(maxWGs >= 1);
+  auto local_range = sycl::range<Dimensions>(Dims...);
+  auto global_range = local_range;
+  global_range[0] *= maxWGs;
+  REQUIRE(global_range.size() == local_range.size() * maxWGs);
+  auto nd_range = sycl::nd_range<Dimensions>{global_range, local_range};
+  auto props = sycl::ext::oneapi::experimental::properties{
+      sycl::ext::oneapi::experimental::use_root_sync};
+
+  struct checks {
+    enum {
+      get_group_id,
+      get_local_id,
+      get_group_range,
+      get_local_range,
+      get_max_local_range,
+      get_group_linear_id,
+      get_local_linear_id,
+      get_group_linear_range,
+      get_local_linear_range,
+      leader,
+      get_child_group___root_group,
+      get_child_group___group,
+      COUNT,
+    };
+  };
+
+  using CheckResults = bool[checks::COUNT];
+  sycl::buffer<CheckResults, Dimensions> results_buffer{global_range};
+
+  q.submit([&](sycl::handler& cgh) {
+    sycl::accessor acc{results_buffer, cgh, sycl::write_only};
+    cgh.parallel_for<KernelName>(nd_range, props, [=](auto it) {
+      auto results = acc[it.get_global_id()];
+
+      auto root = it.ext_oneapi_get_root_group();
+      results[checks::get_group_id] =
+          root.get_group_id() == sycl::range<Dimensions>();
+      results[checks::get_local_id] = root.get_local_id() == it.get_global_id();
+      results[checks::get_group_range] =
+          root.get_group_range() == (1 | sycl::range<Dimensions>());
+      results[checks::get_local_range] =
+          root.get_local_range() == it.get_global_range();
+      results[checks::get_max_local_range] =
+          root.get_max_local_range() == root.get_local_range();
+      results[checks::get_group_linear_id] = root.get_group_linear_id() == 0;
+      results[checks::get_local_linear_id] =
+          root.get_local_linear_id() == it.get_global_linear_id();
+      results[checks::get_group_linear_range] =
+          root.get_group_linear_range() == root.get_group_range().size();
+      results[checks::get_local_linear_range] =
+          root.get_local_linear_range() == root.get_local_range().size();
+      results[checks::leader] = root.leader() == (root.get_local_id() == 0);
+
+      auto g = sycl::ext::oneapi::experimental::get_child_group(root);
+      auto sg = sycl::ext::oneapi::experimental::get_child_group(g);
+      results[checks::get_child_group___root_group] = g == it.get_group();
+      results[checks::get_child_group___group] = sg == it.get_sub_group();
+    });
+  });
+  q.wait();
+
+  CheckResults results = {};
+  sycl::accessor acc = results_buffer.get_host_access();
+  for (int check = 0; check < checks::COUNT; check++) {
+    bool passed = true;
+    for (const auto& it : acc) {
+      passed &= it[check];
+    }
+    results[check] = passed;
+  }
+
+  using rg = sycl::ext::oneapi::experimental::root_group<Dimensions>;
+
+  // type aliases
+  CHECK(std::is_same_v<typename rg::id_type, sycl::id<Dimensions>>);
```

Comment:
I wonder what is a strategy/preferred way of handling type checks in CTS? Should such checks be implemented using a `static_assert` instead to allow earlier detection of bugs?